### DISCUSSION TITLE
关于NET版本的疑问

### DISCUSSION THREAD

**Discussion by CrazyZhang666** [BYJRK/DotNet-Discussions#33] [Q&A]
## 疑问

C:\Program Files\dotnet\packs

这个路径下是NET安装位置，我的是随VS2022更新他自动安装NET版本的，

我发现每次更新都会多一个版本，比如最开始我只有NET6.0.11，现在是6.0.12，6.0.13 ...  7.0.3 版本很多

我们自己在创建项目的时候，只能选择NET6 NET7这些，不能具体到某些版本号 （我可能是受NET Framework 4.5 4.6 4.7 4.8 这种版本命名影响）

我想知道这些NET6.0.11 6.0.12 6.0.13 等等等 有啥区别？假设用户电脑是 NET6.0.11 另一个是NET6.0.12 貌似都可以用NET6写的程序，但是有时候会不行

如果是NET Framework 你用NET4.8写的 如果你安装的NET4.7 那么就会打不开，但是NET6貌似就没这个限制，只要大版本号一样就行

> 说的比较乱，见谅。我主要想表达这些疑惑，因为其他语言的框架一般都是有明确的版本区别的

![image](https://user-images.githubusercontent.com/28080853/219952355-f0d393af-bb70-49a1-92a8-e84647f8544f.png)

## 番外

可以出一期 VS2022 智能提示改中文的视频吗？我找到了这个链接，微软官方NET5.0就停更了中文提示
https://www.cnblogs.com/internalnet/p/16185298.html

```
dotnet tool install -g islocalizer

islocalizer install auto -m net6.0 -l zh-cn -cc None
islocalizer install auto -m net7.0 -l zh-cn -cc None
```

**Comment by KiyanYang** [comment]
可以看一下文档 https://learn.microsoft.com/dotnet/core/releases-and-support ，.net 6 是因为就只发布了这种的版本，之后之间就到下一个大版本了，没有 .net 6.1 这种的，可以参考 .net core 3.1

**Comment by BYJRK** [comment]
.NET 5 以来，目前只有大版本更新和小版本补丁升级，并没有类似 Framework 那样的 4.6、4.7、4.8，以及 Core 的 3、3.1 等。

以 .NET 6 为例，它就有过多次小版本更新，比如 6.0.203、6.0.302 等，以及目前最新的 6.0.406。至于 6.0.14，这个是 Runtime 的版本。
.NET 7 也有小版本更新，比如目前最新的 .NET 7.0.3 等。但是它们之前并没有任何语法等方面的区别，也不会引入新特性，只是会修复之前版本的 bug。

**Comment by Dynesshely** [comment]
dotnet sdk 以及 runtime 大部分均遵循 semver 语义版本控制

您图中的 6.0.11, 6.0.12, 6.0.13, 6.0.14, 变更的只有 patch 版本, 表示这些更新不涉及关键接口, api, 用法等变化, 可能是安全更新, 错误修复等, 所以可以向下兼容同一个大版本的 dotnet 程序

您无需关心 patch 版本变更, 您只用在意 dotnet 大版本选择

您的 dotnet 程序在运行时, 如果依赖系统安装的 dotnet 运行时, 应该会选择同一大版本下最新版本号的运行时来执行
